sdk/go/pulumi/context.go

```diff
@@ -955,3 +1018,13 @@ func (ctx *Context) RegisterResourceOutputs(resource Resource, outs Map) error {
 func (ctx *Context) Export(name string, value Input) {
 	ctx.exports[name] = value
 }
+
+// RegisterStackTransformation adds a transformation to all future resources constructed in this Pulumi stack.
+func (ctx *Context) RegisterStackTransformation(t ResourceTransformation) error {
+	if ctx.stack == nil {
+		return errors.New("the root stack resource was referenced before it was initialized")
+	}
```

Comment:
Is it possible for this to happen in the Go SDK?

Reply:
I don't think so, but having the extra safety check here probably doesn't hurt.

Reply:
Going to remove this check as it seems unnecessary.

Reply:
Going to remove this check since it seems unnecessary.